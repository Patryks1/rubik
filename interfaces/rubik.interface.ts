export type Color = 'red' | 'green' | 'blue' | 'yellow' | 'white' | 'orange';
export type FaceDirection = 'F' | 'L' | 'R' | 'U' | 'D' | 'B';

export interface IMoveInstruction {
  toFace: FaceDirection;
  toTileIndex: number;
  fromFace: FaceDirection;
  fromTileIndex: number;
}

export interface IMoves {
  [x: string]: IMoveInstruction[];
}

export interface IRubikFace {
  tiles: Color[];
  faceDirection: FaceDirection;
}
