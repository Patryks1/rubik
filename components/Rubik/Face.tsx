import React, { FC } from 'react';
import { Color } from '../../interfaces';
import Tile from './Tile';

type Props = {
  tiles: Color[];
};

const Face: FC<Props> = ({ tiles }) => (
  <div className='flex flex-wrap w-[240px] h-[240px]'>
    {tiles.map((color, index) => (
      <Tile key={index} color={color} index={index} />
    ))}
  </div>
);

export default Face;
