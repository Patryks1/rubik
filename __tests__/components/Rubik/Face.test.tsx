import { render, cleanup, screen } from '@testing-library/react';
import React from 'react';
import Face from '@/components/Rubik/Face';
import { Color } from 'interfaces';

describe('Face', () => {
  afterAll(() => {
    cleanup();
  });

  it('renders Face', () => {
    const tiles: Color[] = Array(9).fill('red');
    const { container } = render(<Face tiles={tiles} />);
    expect(container).toMatchSnapshot();
  });

  it('renders 3x3 grid', () => {
    const tiles: Color[] = Array(9).fill('red');
    render(<Face tiles={tiles} />);

    const cells = screen.getAllByTestId('tile-cell');

    expect(cells.length).toBe(9);
  });
});
