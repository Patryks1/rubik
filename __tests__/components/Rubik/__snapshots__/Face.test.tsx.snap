// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Face renders Face 1`] = `
<div>
  <div
    class="flex flex-wrap w-[240px] h-[240px]"
  >
    <div
      class="flex-[0_0_80px] h-[80px] w-[80px] border border-black"
      data-testid="tile-cell"
      style="background-color: red;"
    >
      0
    </div>
    <div
      class="flex-[0_0_80px] h-[80px] w-[80px] border border-black"
      data-testid="tile-cell"
      style="background-color: red;"
    >
      1
    </div>
    <div
      class="flex-[0_0_80px] h-[80px] w-[80px] border border-black"
      data-testid="tile-cell"
      style="background-color: red;"
    >
      2
    </div>
    <div
      class="flex-[0_0_80px] h-[80px] w-[80px] border border-black"
      data-testid="tile-cell"
      style="background-color: red;"
    >
      3
    </div>
    <div
      class="flex-[0_0_80px] h-[80px] w-[80px] border border-black"
      data-testid="tile-cell"
      style="background-color: red;"
    >
      4
    </div>
    <div
      class="flex-[0_0_80px] h-[80px] w-[80px] border border-black"
      data-testid="tile-cell"
      style="background-color: red;"
    >
      5
    </div>
    <div
      class="flex-[0_0_80px] h-[80px] w-[80px] border border-black"
      data-testid="tile-cell"
      style="background-color: red;"
    >
      6
    </div>
    <div
      class="flex-[0_0_80px] h-[80px] w-[80px] border border-black"
      data-testid="tile-cell"
      style="background-color: red;"
    >
      7
    </div>
    <div
      class="flex-[0_0_80px] h-[80px] w-[80px] border border-black"
      data-testid="tile-cell"
      style="background-color: red;"
    >
      8
    </div>
  </div>
</div>
`;
